infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SenderBufferManager.java

```diff
@@ -175,20 +175,20 @@ public void sendWithBuffering(List<LogReplicationEntry> dataToSend) {
 
     /**
      * Resend the messages in the queue if they have timed out.
-     * @return false, if an entry has been resent MAX_RETRY.
-     *         true, otherwise
      */
     public LogReplicationEntry resend() {
-        //Enforce a resend or not
         LogReplicationEntry ack = null;
         boolean force = false;
         try {
             ack = processAcks();
-        } catch (TimeoutException e) {
-            log.warn("Caught a timeout exception ", e);
-            force = true;
+        } catch (ExecutionException ee) {
```

Comment:
I think we still need the catch block for `TimeoutException`.

There are two kinds of `TimeoutException`:
1. From the actual sendMsg's CF, this type will be wrapped with the `ExecutionException `
2. From anyOf(cfs).get(long timeout, TimeUnit unit)

Reply:
Yes, you're right. Added a catch on TimeoutException as well...